homeworks/07-mars-viewer/src/store/sols.ts

```diff
@@ -0,0 +1,82 @@
+import {createAsyncThunk, createSlice, PayloadAction} from "@reduxjs/toolkit";
+
+export interface IImage {
+	id: number,
+	url: string,
+	cameraName: string,
+	roverName: string
+}
+
+export interface IImages {
+	[key: number]: IImage[]
+}
+
+interface ISols {
+	selectedSol: number,
+	isLoading: boolean,
+	images: IImages,
+	favorite: number[]
+}
+
+interface IResponse {
+	photos: Array<{
+		id: number,
+		img_src: string,
+		camera: { full_name: string },
+		rover: { name: string }
+	}>
+}
+
+const initialState: ISols = {
+	selectedSol: 1,
+	isLoading: false,
+	images: {},
+	favorite: []
+}
+
+export const loadImages: any = createAsyncThunk<IImage[], number, {}>(
+	"sols/loadImages",
+	async (sol) => {
+		const response = await fetch(`https://api.nasa.gov/mars-photos/api/v1/rovers/perseverance/photos?sol=${sol}&api_key=etwdnixT9DMgG9WbdSmHEqbReflMHWMEYUosQOyf`);
```

Comment:
`create-react-app` allows passing env variables without having to additionally configure bundler to pass them. For that you just need to create `.env` file in root directory and add variable with `REACT_APP_` prefix https://create-react-app.dev/docs/adding-custom-environment-variables/